upb/def.c

```diff
@@ -1115,27 +1185,33 @@ static void make_layout(symtab_addctx *ctx, const upb_msgdef *m) {
   upb_msg_oneof_iter oit;
   size_t hasbit;
   size_t field_count = upb_msgdef_numfields(m);
-  size_t submsg_count = 0;
-  const upb_msglayout **submsgs;
+  size_t sublayout_count = 0;
+  upb_msglayout_sub *subs;
   upb_msglayout_field *fields;
 
   memset(l, 0, sizeof(*l) + sizeof(_upb_fasttable_entry));
 
   /* Count sub-messages. */
   for (size_t i = 0; i < field_count; i++) {
     if (upb_fielddef_issubmsg(&m->fields[i])) {
-      submsg_count++;
+      sublayout_count++;
     }
   }
 
   fields = symtab_alloc(ctx, field_count * sizeof(*fields));
-  submsgs = symtab_alloc(ctx, submsg_count * sizeof(*submsgs));
+  subs = symtab_alloc(ctx, sublayout_count * sizeof(*subs));
 
   l->field_count = upb_msgdef_numfields(m);
   l->fields = fields;
-  l->submsgs = submsgs;
+  l->subs = subs;
   l->table_mask = 0;
 
+  if (upb_msgdef_extrangecount(m) > 0) {
```

Comment:
in particular the compact naming is hard here.  As a reader I see "upb message definition extra nge" oh shit backtrack

Reply:
I can totally see why this harms readability.

To give some background, I took the naming convention from Lua: https://www.lua.org/manual/5.3/#index

I now wish I had used `upb_UpperCamelCase_Operation`, as I've done in Python/upb: https://github.com/protocolbuffers/upb/blob/master/python/descriptor_pool.c

I would be happy to undertake a renaming that moves the upb C API to this convention, once the current batch of work is landed.